src/UI/templates/js/MainControls/mainbar.js

```diff
@@ -0,0 +1,301 @@
+			,_cls_single_slate = false //class of one single slate, will be set on registerSignals
+			,_cls_slate_engaged = false //engaged class of a slate, will be set on registerSignals
+			,_cls_mainbar = 'il-mainbar';
+		;
+
+		var registerSignals = function (
+			component_id,
+			entry_signal,
+			close_slates_signal,
+			tools_signal,
+			tools_removal_signal
+		) {
+			id = component_id;
+			_cls_single_slate = il.UI.maincontrols.slate._cls_single_slate;
+			_cls_slate_engaged = il.UI.maincontrols.slate._cls_engaged;
+
+			$(document).on(entry_signal, function(event, signalData) {
+				onClickEntry(event, signalData);
+				return false;
+			});
+			$(document).on(close_slates_signal, function(event, signalData) {
+				onClickDisengageAll(event, signalData);
+				return false;
+			});
+			$(document).on(tools_signal, function(event, signalData) {
+				onClickToolsEntry(event, signalData);
+				return false;
+			});
+			$(document).on(tools_removal_signal, function(event, signalData) {
+				onClickToolRemoval(event, signalData);
+				return false;
+			});
+		};
+
+		var initActive = function(component_id) {
+			id = component_id;
+			var btn = _getAllButtons()
+				.filter('.' + _cls_btn_engaged);
+			_disengageButton(btn);
+			btn.click();
+		}
+
+		var onClickEntry = function(event, signalData) {
+			var btn = signalData.triggerer;
+			if(_isEngaged(btn)) {
+				if(! _isToolButton(btn)) {
+					_disengageButton(btn);
+					_setPageSlatesActive(false);
+				}
+			} else {
+				_disengageAllButtons(); //reset, so that only _one_ is active
+				if(_isToolButton(btn)) {
+					_disengageAllToolButtons();
+				}
+				_disengageAllSlates();
+				_engageButton(btn);
+				_setPageSlatesActive(true);
+				if(_isToolButton(btn)) {
+					_setToolsActive(true);
+					_engageButton(_getToolsButton());
+				} else {
+					_setToolsActive(false);
+					_disengageButton(_getToolsButton());
+				}
+			}
+		};
+
+		var onClickDisengageAll = function(event, signalData) {
+			_disengageAllButtons();
+			_disengageAllSlates();
+			_setPageSlatesActive(false);
+			_setToolsActive(false);
+		};
+
+		var onClickToolsEntry = function(event, signalData) {
+			var btn = signalData.triggerer,
+				active_tool_button;
+
+			if(_isEngaged(btn)) {
+				_setPageSlatesActive(false);
+				_setToolsActive(false);
+				_disengageButton(btn);
+			} else {
+				_disengageAllButtons();
+				_setPageSlatesActive(true);
+				_setToolsActive(true);
+				_engageButton(btn);
+				_disengageAllSlates();
+
+				if(_isAnyToolActive()) {
+					active_tool_btn = _getAllToolButtons()
+						.filter('.' + _cls_btn_engaged)[0];
+				} else {
+					active_tool_btn = _getAllToolButtons()[0];
+				}
+				active_tool_btn.click();
+			}
+		};
+
+		var onClickToolRemoval = function(event, signalData) {
+			var search = '#' + id + ' .' + _cls_toolentries_wrapper + ' .btn',
+				active_tool_btn = $(search).filter(' .' + _cls_btn_engaged),
+				remaining;
```

Comment:
The indentation screws with the readability here. I needed some time to see that this is actually a variable definition instead of a usage of `active_tool_btn`.